std/asys/FileSystem.hx

```diff
@@ -0,0 +1,222 @@
+package asys;
+
+import haxe.Error;
+import haxe.io.Bytes;
+import haxe.io.FilePath;
+import asys.io.*;
+
+typedef FileReadStreamCreationOptions = {
+	?flags:FileOpenFlags,
+	?mode:FilePermissions
+} &
+	asys.io.FileReadStream.FileReadStreamOptions;
+
+/**
+	This class provides methods for synchronous operations on files and
```

Comment:
A bit weird to have a completely synchronous API in `asys` package

Reply:
Well, we're not going to have multiple packages for this, and we cannot merge it with `sys` easily (more so due to `asys.io.File` incompatibilities with `sys.io.File`).

Reply:
But

> The types in asys will only be available on libuv-backed system targets 

while `asys.FileSystem` and such are perfectly doable without libuv.

> we're not going to have multiple packages for this

I think we should split it. Maybe `io.sync` and `io.async` or whatever better names?